### ISSUE TITLE
TFJob will not run if the TFJob CRD includes scope: Namespaced

### ISSUE THREAD

**Issue by kkasravi** [kubeflow/kubeflow#1606]
@jlewi There is a subtle issue in tf-job-operator

this works and results in tf-job-simple running
```
---
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  labels:
    ksonnet.io/component: tf-job-operator
  name: tfjobs.kubeflow.org
spec:
  group: kubeflow.org
  names:
    kind: TFJob
    plural: tfjobs
    singular: tfjob
  validation:
    openAPIV3Schema:
      properties:
        spec:
          properties:
            tfReplicaSpecs:
              properties:
                Chief:
                  properties:
                    replicas:
                      maximum: 1
                      minimum: 1
                      type: integer
                PS:
                  properties:
                    replicas:
                      minimum: 1
                      type: integer
                Worker:
                  properties:
                    replicas:
                      minimum: 1
                      type: integer
  version: v1alpha2
```
and this doesn't (difference is the inclusion of the scope attribute).
There are no errors that I can see and tf-job-simple never runs.

```
---
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  labels:
    ksonnet.io/component: tf-job-operator
  name: tfjobs.kubeflow.org
spec:
  group: kubeflow.org
  names:
    kind: TFJob
    plural: tfjobs
    singular: tfjob
  scope: cluster
  validation:
    openAPIV3Schema:
      properties:
        spec:
          properties:
            tfReplicaSpecs:
              properties:
                Chief:
                  properties:
                    replicas:
                      maximum: 1
                      minimum: 1
                      type: integer
                PS:
                  properties:
                    replicas:
                      minimum: 1
                      type: integer
                Worker:
                  properties:
                    replicas:
                      minimum: 1
                      type: integer
  version: v1alpha2
```

Since the default for CustomResourceDefinition scope is [Namespaced](https://godoc.org/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions) is seems like this would be a bug. I couldn't find in tf-operator repo where this is referenced.

**Comment by jlewi** [comment]
I think this is WAI is intended. 

Scope refers to whether the TFJob should be a namespace or not namespace scoped object.
I think it makes sense for TFJob's to always be namespace scoped. Meaning that every TFJob belongs to a namespace.

Our tests assume its namespace scoped; so when we query for the status of the job we always specify the namespace.

If this is in regards to #1535, I suspect what might have happened is the refactor ended up defining the TFJob CR as a cluster scope resource. The E2E tests then created the TFJobs but the test infrastructure then failed to get TFJob status because it was looking for the job in a particular namespace and not finding it because the job was created as a cluster level resource.

**Comment by kkasravi** [comment]
Thanks @jlewi 

TL;DR 
if [tfJobVersion default](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/prototypes/tf-job-operator.jsonnet#L11) is v1alpha2 then 
  - the [crdv1alpha2](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/tf-job-operator.libsonnet#L44) manifest should include scope: namespaced.
  - the [deploymentScope parameter](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/prototypes/tf-job-operator.jsonnet#L12) default should be namespace.
if the tfJobVersion default is v1alpha1 then 
  - the [crd](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/tf-job-operator.libsonnet#L27) manifest should include scope: cluster.
  -  the [deploymentScope parameter](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/prototypes/tf-job-operator.jsonnet#L12) default should be cluster.

In both cases the TFJob tf-job-simple should run.

---

This doesn't involve #1535 per se. The [default params](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/prototypes/tf-job-operator.jsonnet#L11) of tfJobVersion, deploymentScope and deploymentNamespace on master aren't compatible. If we're using tfJobVersion of v1alpha2 then the deploymentScope should be namespace not cluster and the TFJob CRD should have an attribute of scope: Namespaced. But both TFJob crds: [crd](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/tf-job-operator.libsonnet#L27) and [crdv1alpha2](https://github.com/kubeflow/kubeflow/blob/master/kubeflow/core/tf-job-operator.libsonnet#L44) leave out the scope attribute.

**Comment by kkasravi** [comment]
@jlewi - update the CRD to include scope: Namespaced in PR #1535

**Comment by jlewi** [comment]
@kkasravi 

I think its fine to drop support for v1alpha1 at this point if that simplifies things.

**Comment by kkasravi** [comment]
this has been resolved elsewhere
